src/astarte_device.c

```diff
@@ -77,12 +77,8 @@ static void on_disconnected(astarte_device_handle_t device);
 static void on_incoming(
     astarte_device_handle_t device, char *topic, int topic_len, char *data, int data_len);
 static void on_certificate_error(astarte_device_handle_t device);
-#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
```

Comment:
Why did you remove this check? Is it still valid on `4.2`?

Reply:
Those macros had incorrect arguments, they should have been 
```
#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 0, 0)
```